src/main/java/io/reactivex/internal/operators/parallel/ParallelCollect.java

```diff
@@ -0,0 +1,165 @@
+/**
+ * Copyright 2016 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in
+ * compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is
+ * distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See
+ * the License for the specific language governing permissions and limitations under the License.
+ */
+
+package io.reactivex.internal.operators.parallel;
+
+import java.util.concurrent.Callable;
+
+import org.reactivestreams.*;
+
+import io.reactivex.exceptions.Exceptions;
+import io.reactivex.functions.BiConsumer;
+import io.reactivex.internal.subscribers.DeferredScalarSubscriber;
+import io.reactivex.internal.subscriptions.*;
+import io.reactivex.parallel.ParallelFlowable;
+import io.reactivex.plugins.RxJavaPlugins;
+
+/**
+ * Reduce the sequence of values in each 'rail' to a single value.
+ *
+ * @param <T> the input value type
+ * @param <C> the collection type
+ */
+public final class ParallelCollect<T, C> extends ParallelFlowable<C> {
+
+    final ParallelFlowable<? extends T> source;
+
+    final Callable<C> initialCollection;
+
+    final BiConsumer<C, T> collector;
+
+    public ParallelCollect(ParallelFlowable<? extends T> source,
+            Callable<C> initialCollection, BiConsumer<C, T> collector) {
+        this.source = source;
+        this.initialCollection = initialCollection;
+        this.collector = collector;
+    }
+
+    @Override
+    public void subscribe(Subscriber<? super C>[] subscribers) {
+        if (!validate(subscribers)) {
+            return;
+        }
+
+        int n = subscribers.length;
```

Comment:
In long methods reader has to spend extra time to check that it's not modified anywhere, but ok